lib/targets/cuda/quda_api.cpp

```diff
@@ -123,10 +126,10 @@ namespace quda {
       auto search = cache.find(func);
       if (search == cache.end()) {
         cache.insert(func);
-        qudaFuncSetAttribute(func, cudaFuncAttributePreferredSharedMemoryCarveout, (int)cudaSharedmemCarveoutMaxShared);
+        cudaFuncSetAttribute(func, cudaFuncAttributePreferredSharedMemoryCarveout, (int)cudaSharedmemCarveoutMaxShared);
```

Comment:
Why these changes?

Reply:
This is internal to CUDA in a target specific file now. So we can have it either way. I don't think HIP has this carveaout attribute, so it doesn't need to be abstracted

Reply:
Indeed, but please revert this change.  I still want this abstracted for (future) profiling reasons.

Reply:
OK. Reverted in e9306c8d8 -- tho it is odd because it is a qudaXXXX() call that is not in quda_api.h and takes cudaFuncAttributes as params...

Reply:
Thanks.  Yes I know it might look odd 😉 .  It's to give me an easy handle for profiling the overhead of this function.